### ISSUE TITLE
Async Jquery searches

### ISSUE THREAD

**Issue by djstreat** [simsong/NativeFileList#4]
Currently, it appends the search to the end of the actual search. This can be slow; we should experiment with it injecting some Jquery that does the search asynchronously and pops in the results.

**Comment by djstreat** [comment]
@simsong I ran into some trouble today with trying to get configuration settings set up. I've been looking at the documentation for config settings [here](https://www.mediawiki.org/wiki/Manual:Extension_registration#Configs_(Your_extension/skins_settings)). The new branch `features/async_jquery_search` should have the settings there, but when running the update, `$nflDBPrefix` gives: `PHP Notice:  Undefined variable: nflDBPrefix in /var/www/mediawiki-1.33.1/extensions/NativeFileList/includes/Hooks.php on line 115`

Not sure what is causing this problem, but am still looking into it.

**Comment by simsongcensus** [comment]
Do you need to define it with a default value at the top of the Hooks.php file?

**Comment by djstreat** [comment]
Figured it out before I left on Friday: I had to declare it as global before I used it.

**Comment by simsongcensus** [comment]
Yes! It has to be set *somewhere*!

Sent from my iPhone

On Nov 25, 2019, at 8:40 AM, djstreat <notifications@github.com> wrote:

﻿

Figured it out before I left on Friday: I had to declare it as global before I used it.

**Comment by djstreat** [comment]
Just a quick note on this front:

I seem to be running into scenarios where when trying to insert scripts that use jquery, my scripts end up running before jquery is loaded. This is because mediawiki asynchronously loads jquery. It then becomes difficult to figure out how to run my code, since calling `$.ajax()` will return a reference error, even when set to a timeout function.

**Comment by simsong** [comment]
That’s wild. How about not using Jquery?


----
Sent from my phone.

> On Nov 26, 2019, at 5:16 PM, djstreat <notifications@github.com> wrote:
> 

﻿
Just a quick note on this front:

I seem to be running into scenarios where when trying to insert scripts that use jquery, my scripts end up running before jquery is loaded. This is because mediawiki asynchronously loads jquery. It then becomes difficult to figure out how to run my code, since calling $.ajax() will return a reference error, even when set to a timeout function.

**Comment by djstreat** [comment]
_somewhat_ of a success with the append after results hook

![Screen Shot 2019-11-26 at 3 21 44 PM](https://user-images.githubusercontent.com/57365532/69669837-9fac5e80-1060-11ea-8106-529eec4b801b.png)

**Comment by simsongcensus** [comment]
Really good. If you create a [Talk] page, does the color change? That is, are you creating these the correct way, or are you still using my broken code?


Sent from my iPad

On Nov 26, 2019, at 9:23 PM, djstreat <notifications@github.com<mailto:notifications@github.com>> wrote:


somewhat of a success with the append after results hook

[Screen Shot 2019-11-26 at 3 21 44 PM]<https://user-images.githubusercontent.com/57365532/69669837-9fac5e80-1060-11ea-8106-529eec4b801b.png>

**Comment by djstreat** [comment]
Ah, the color does not change, although the page does exist. I will work on fixing that next.

**Comment by simsongcensus** [comment]
Yes, apparently there is a proper way to put up the Talk link. I am not following it. If you look in the notes on my Stackoverflow pages, you should find it.

Sent from my iPad

On Nov 26, 2019, at 10:25 PM, djstreat <notifications@github.com<mailto:notifications@github.com>> wrote:


Ah, the color does not change, although the page does exist. I will work on fixing that next.
